src/main/scala/io/citrine/lolo/trees/splits/BoltzmannSplitter.scala

```diff
@@ -0,0 +1,226 @@
+package io.citrine.lolo.trees.splits
+
+import io.citrine.lolo.trees.impurity.VarianceCalculator
+import io.citrine.lolo.trees.splits.BoltzmannSplitter.SplitterResult
+
+import scala.collection.mutable
+import scala.util.Random
+
+/**
+  * Find a split for a regression problem
```

Comment:
What are the expected benefits of Boltzmann trees? Why is introducing more randomness into which split is picked expected to improve model performance? Would temperature be a hyperparameter that automatically gets optimized?

Reply:
Added